docs/contrib/workflow.md

```diff
@@ -0,0 +1,232 @@
+start to work on it.
+
+### Design
+
+If you are working on a small feature, fix or improvement you can skip the design step.
+But if you are working on a big feature, especially which changes API, you must
+create a design document first. Examples of design documents can be found [here](../design).
+
+## Development
+
+### Git branches
+
+To make any changes in OpenKilda repository you need to create a git branch.
+There are 2 main branches in the repository:
+
+* `develop` - contains changes which were not yet released
+* `master` - contains last [release](https://github.com/telstra/open-kilda/releases)
+
+Please checkout your branch from `develop` branch.
+
+### Branch name
+
+There are no strict branch name restrictions. The only rule is to make branch name meaningful.
+In practice, we are using the following prefixes for branches:
+
+* `doc/` - for adding/updating the documentation
+* `feature/` - for adding new features
+* `fix/` - for fixing current codebase
+* `test/` - for tests
+* `release` - for releases (only a release manager can create such branches)
+* `hotfix` - for hotfixes which must be released ASAP (only a release manager can create such branches)
+
+After the prefix there must be a very short description of a changes.
+
+Good branch names:
+
+* `doc/contribution_guide`
+* `feature/ha_flow_create`
+* `fix/incorrect_latency_validation`
+* `test/ha_flow_functional_tests`
+* `release-1.2.3`
+* `hotfix-1.2.4`
+
+Bad branch names:
+
+* `guide`
+* `refactoring`
+* `added_tests`
+* `my_mega_fix`
+* `doc/how_to_contribute_in_kilda_if_you_cant_come_up_with_a_short_name_for_a_git_branch`
+
+### Commits
+
+The best practice is to have one commit per one branch. If you need to change
+your code after it was pushed - just amend your new changes into your last commit
+(`git commit --amend`).
+
+If you will, you can have several commits in your branch but please use meaningful
+commit titles for them. First commit must describe a main idea of your pull request.
+Rest commit titles must describe changes which were added after the fist commit.
+
+Good commit titles in a branch:
+
+```
+Added HA flow create operation
+Added unit tests for HA flow create
+Fixed checkstyle issues
+Removed unused files
+```
+
+Bad commit titles in a branch:
+
+```
+Added HA flow create operation
+fix
+fix
+fix
+```
+
+```
+Added HA flow create operation
+Added HA flow create operation
+Added HA flow create operation
+```
+
+```
+Added HA flow create operation
+debug
+Fixed issues after code review
+```
+
+### Commit message
+
+Commit message consist of several parts:
+* Title - one sentence with describes the main idea of your changes
+* Description - several sentences which describes your changes and gives
+  some addition information
+* Tags - helps GitHub to link PRs and issues
+
+When you create a new pull request (PR) from your branch on GitHub it takes
+commit message of your first commit in a branch as PR description.
```

Comment:
Is it? This PR doesn't have a description, but the first commit has a message `Added OpenKilda contribution workflow description`

Reply:
That's true. Github actually takes the description of the commit. But this commit has no description, only a title.
@niksv I would add a description as well to the commit. "This commit introduces a contribution guide to assist developers in contributing to the Open-Kilda project in accordance with its guidelines." (I am starting to love AI's btw)

Reply:
added)